File: platforms/android/keyboard-renderer/src/main/java/app/funput/funput/keyboard/model/KeyboardEnterAction.kt
package app.funput.funput.keyboard.model

/** Visual meaning exposed by the keyboard's context-sensitive Enter key. */
sealed interface KeyboardEnterAction {
    enum class Standard : KeyboardEnterAction {
        NEW_LINE,
        GO,
        SEARCH,
        SEND,
        NEXT,
        DONE,
        PREVIOUS,
    }

    data class Custom(val label: String) : KeyboardEnterAction {
        init {
            require(label.isNotBlank()) { "Custom action label must not be blank" }
        }
    }
}

File: platforms/ios/Packages/FunputKit/Sources/KeyboardLayout/Models/KeyboardEnterAction.swift
import Foundation

public enum KeyboardEnterAction: Hashable, Sendable {
    case newLine
    case go
    case search
    case send
    case next
    case done
    case previous
    case custom(String)

    public static func validatedCustom(_ label: String) -> Self {
        precondition(!label.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty)
        return .custom(label)
    }
}
